frontend/src/app/home/electricity/page.tsx

```diff
@@ -28,21 +28,27 @@ export default function Electricity() {
   const [data, setData] = useState<ElectricityDataProps>([]);
 
   useEffect(() => {
-    async function fetchData() {
-      const temp = await fetch(
-        `http://${process.env.NEXT_PUBLIC_HOST_URL}/service/house/0/electricity`,
-        {
-          next: { revalidate: 60 }, // Revalidate every 60 seconds
-        },
-      );
-      setData(await temp.json());
+    const socket = new WebSocket('ws://localhost:8080/websocket');
+
+    socket.onmessage = (event) => {
+      const newData = JSON.parse(event.data);
+      setData((prevData) => [...prevData, newData])
     }
 
-    fetchData().catch(console.error);
-    const interval = setInterval(() => {
-      fetchData().catch(console.error);
-    }, 5000);
-    return () => clearInterval(interval);
+    // async function fetchData() {
```

Comment:
Isn't the API fetching needed at component mounting? From my understanding (and you did it well), the socket only sends new real-time data, so the past data must be fetched once, right? Please let me know if I'm wrong.